src/ipc/ipc4/helper.c

```diff
@@ -371,17 +371,18 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 		goto free;
 	}
 
-	ret = comp_bind(source, bu);
-	if (ret < 0)
-		goto e_src_bind;
 
 	ret = comp_buffer_connect(sink, sink->ipc_config.core, buffer,
 				  PPL_CONN_DIR_BUFFER_TO_COMP);
 	if (ret < 0) {
 		tr_err(&ipc_tr, "failed to connect internal buffer to sink %d", sink_id);
-		goto e_sink_connect;
+		goto e_src_bind;
```

Comment:
So actually, please keep this as "goto e_sink_connect;" and then
at end of function, have:
```
err:
	comp_unbind(sink, bu);
e_sink_bind:
	comp_unbind(source, bu);
e_src_bind:
	pipeline_disconnect(sink, buffer, PPL_CONN_DIR_BUFFER_TO_COMP);
e_sink_connect:
	pipeline_disconnect(source, buffer, PPL_CONN_DIR_COMP_TO_BUFFER);
```

Reply:
Done.Is it now correct?